src/v7400/object/model.rs

```diff
@@ -57,4 +59,56 @@ impl<'a> ModelHandle<'a> {
                 _ => None,
             })
     }
+
+    /// Gets the root-most model parent of this object in the scene hierarchy.
+    ///
+    /// If this object has no parents, returns this object.
+    ///
+    /// There may be multiple root models in a single scene.
+    pub fn root_model<'b>(&'b self) -> TypedModelHandle<'a> {
+        let mut parent : TypedModelHandle = match self.get_typed() {
+            TypedObjectHandle::Model(o) => o,
+            _ => panic!("ModelHandle should always have TypedObjectHandle::Model type.")
+        };
+
+        while let Some(m) = parent.parent_model() {
+            parent = m;
+        }
+        return parent;
+    }
+
+    /// Returns the local rotation (Lcl Rotation) of this model object, if one is present.
+    pub fn local_rotation(&self) -> Option<Vector3<f64>> {
+        let rotation = self.direct_properties()?.get_property("Lcl Rotation")?;
```

Comment:
This line always fetches the file-wide default local rotation, but is this really your intention?

```suggestion
        let rotation = self.direct_properties()?.get_property("Lcl Rotation")?;
```